spec/micro/README.md

```diff
@@ -0,0 +1,781 @@
+
+If the client and counterparty identifiers are setup correctly, then the correctness and soundness properties of IBC holds. IBC packet flow is guaranteed to succeed. If a user sends a packet with the wrong destination channel, then as we will see it will be impossible for the intended destination to correctly verify the packet thus, the packet will simply time out.
+
+
+### Registering IBC applications on the router
+
+The IBC router contains a mapping from a reserved application port and the supported versions of that application as well as a mapping from channelIdentifiers to channels.
+
+```typescript
+type IBCRouter struct {
+    apps: portID -> [Version]
+    callbacks: portID -> [Callback]
+    clients: clientId -> Client
+}
+```
+
+### Packet Flow through the Router
+
+```typescript
+function sendPacket(
+    sourcePort: Identifier,
+    sourceChannel: Identifier,
+    destPort: Identifier,
+    timeoutHeight: Height,
+    timeoutTimestamp: uint64,
+    packetData: []byte
+): uint64 {
+    // in this specification, the source channel is the clientId
+    client = router.clients[packet.sourceChannel]
+    assert(client !== null)
+
+    // disallow packets with a zero timeoutHeight and timeoutTimestamp
+    assert(timeoutHeight !== 0 || timeoutTimestamp !== 0)
+
+    // check that the timeout height hasn't already passed in the local client tracking the receiving chain
+    latestClientHeight = client.latestClientHeight()
+    assert(timeoutHeight === 0 || latestClientHeight < timeoutHeight)
+
+    // IBC only commits sourcePort, sourceChannel, sequence in the commitment path
+    // and packet data, and timeout information in the value
+    // For IBC Lite, we can't automatically retrieve the destination port since we don't have an actual stored channel
+    // in order to get around this, if sourcePort and destinationPort are the same we will leave the path as-is.
+    // if the ports are different than we must append the port ids together
+    // so the receiver can verify the requested destination port
+    commitPort = sourcePort
+    if sourcePort != destPort {
+        commitPort = sourcePort + "/" + destPort
+    }
+
+    // if the sequence doesn't already exist, this call initializes the sequence to 0
+    sequence = channelStore.get(nextSequenceSendPath(commitPort, sourceChannel))
+    
+    // store commitment to the packet data & packet timeout
+    channelStore.set(
+      packetCommitmentPath(commitPort, sourceChannel, sequence),
+      hash(hash(data), timeoutHeight, timeoutTimestamp)
+    )
+
+    // increment the sequence. Thus there are monotonically increasing sequences for packet flow
+    // from sourcePort, sourceChannel pair
+    channelStore.set(nextSequenceSendPath(commitPort, sourceChannel), sequence+1)
+
```

Comment:
This should be making an application callback I think. This is because:
1. Even if we want the packet lifecycle to begin in the IBC app. There is nothing wrong with the application receiving a callback. In fact, this is necessary to receive a reply in the Actor model of CosmWasm.
2. Might make multipacket lifecycle easier to implement